File: ocmsdk/src/main/java/com/gigigo/orchextra/core/data/api/mappers/elementcache/ApiElementCacheRenderMapper.java
package com.gigigo.orchextra.core.data.api.mappers.elementcache;

import com.gigigo.ggglib.mappers.ExternalClassToModelMapper;
import com.gigigo.orchextra.core.data.api.dto.article.ApiArticleElement;
import com.gigigo.orchextra.core.data.api.dto.elementcache.ApiElementCacheRender;
import com.gigigo.orchextra.core.data.api.mappers.article.ApiArticleElementMapper;
import com.gigigo.orchextra.core.domain.entities.article.base.ArticleElement;
import com.gigigo.orchextra.core.domain.entities.elementcache.ElementCacheRender;
import com.gigigo.orchextra.core.domain.entities.elementcache.VideoFormat;
import java.util.ArrayList;
import java.util.List;

public class ApiElementCacheRenderMapper
    implements ExternalClassToModelMapper<ApiElementCacheRender, ElementCacheRender> {

  private final ApiArticleElementMapper apiArticleElementMapper;
  private final FederatedAuthorizationDataMapper federatedAuthorizationDataMapper;

  public ApiElementCacheRenderMapper(ApiArticleElementMapper apiArticleElementMapper,
      FederatedAuthorizationDataMapper federatedAuthorizationDataMapper) {
    this.apiArticleElementMapper = apiArticleElementMapper;
    this.federatedAuthorizationDataMapper = federatedAuthorizationDataMapper;
  }

  @Override public ElementCacheRender externalClassToModel(ApiElementCacheRender data) {
    ElementCacheRender model = new ElementCacheRender();

    model.setContentUrl(data.getContentUrl());
    model.setUrl(data.getUrl());

    model.setTitle(data.getTitle());

    model.setSource(data.getSource());
    if (data.getFormat() != null) {
      model.setFormat(VideoFormat.convertStringToType(data.getFormat()));
    }

    model.setSchemeUri(data.getSchemeUri());

    List<ArticleElement> articleElementList = new ArrayList<>();
    if (data.getElements() != null) {
      for (ApiArticleElement apiArticleElement : data.getElements()) {
        ArticleElement articleElement =
            apiArticleElementMapper.externalClassToModel(apiArticleElement);

        if (articleElement != null) {
          articleElementList.add(articleElement);
        }
      }

      //ArticleButtonElement articleButtonElement = new ArticleButtonElement();
      //articleButtonElement.setType(ArticleButtonType.IMAGE);
      //articleButtonElement.setSize(ArticleButtonSize.MEDIUM);
      //articleButtonElement.setElementUrl("element/article/THE-7-BEST-COLLABORATIONS-OF-2016-SkWhR4W4g");
      //articleButtonElement.setText("Ver en mapa");
      //articleButtonElement.setTextColor("#FF4422");
      //articleButtonElement.setBgColor("#4422FF");
      //articleButtonElement.setImageUrl("https://i.ytimg.com/vi/kzvdmSA13Nc/maxresdefault.jpg");
      //articleElementList.add(articleButtonElement);
    }

    model.setFederatedAuth(
        federatedAuthorizationDataMapper.externalClassToModel(data.getFederatedAuth()));

    model.setElements(articleElementList);

    return model;
  }
}

File: ocmsdk/src/main/java/com/gigigo/orchextra/core/data/api/mappers/elementcache/ApiElementCacheRenderMapper.kt
package com.gigigo.orchextra.core.data.api.mappers.elementcache

import android.util.Log
import com.gigigo.ggglib.mappers.ExternalClassToModelMapper
import com.gigigo.orchextra.core.data.api.dto.elementcache.ApiElementCacheRender
import com.gigigo.orchextra.core.data.api.mappers.article.ApiArticleElementMapper
import com.gigigo.orchextra.core.domain.entities.article.base.ArticleElement
import com.gigigo.orchextra.core.domain.entities.article.base.ArticleElementRender
import com.gigigo.orchextra.core.domain.entities.elementcache.ElementCacheRender
import com.gigigo.orchextra.core.domain.entities.elementcache.VideoFormat
import java.util.ArrayList

class ApiElementCacheRenderMapper(private val apiArticleElementMapper: ApiArticleElementMapper,
    private val federatedAuthorizationDataMapper: FederatedAuthorizationDataMapper) : ExternalClassToModelMapper<ApiElementCacheRender, ElementCacheRender> {

  override fun externalClassToModel(data: ApiElementCacheRender): ElementCacheRender {
    val time = System.currentTimeMillis()

    val model = ElementCacheRender()

    with(model) {
      contentUrl = data.contentUrl
      url = data.url
      title = data.title
      source = data.source
      schemeUri = data.schemeUri
      federatedAuth = federatedAuthorizationDataMapper.externalClassToModel(data.federatedAuth)
    }

    data.format?.let {
      model.format = VideoFormat.convertStringToType(data.format)
    }

    val articleElementList = ArrayList<ArticleElement<ArticleElementRender>>()
    if (data.elements != null) {
      for (apiArticleElement in data.elements) {
        val articleElement = apiArticleElementMapper.externalClassToModel(apiArticleElement)

        articleElement?.let {
          articleElementList.add(articleElement)
        }
      }
    }

    model.elements = articleElementList

    val currentTime = System.currentTimeMillis() - time
    Log.v("TT - ApiElemCacheRender", ("" + currentTime/1000))

    return model
  }
}
